src/components/PopupWithForm.js

```diff
@@ -23,10 +27,26 @@ export default class PopupWithForm extends Popup {
       e.preventDefault();
       const inputValues = this._getInputValues();
       this._handleFormSubmit(inputValues);
-      this.close();
     });
   }
 
+  setLoadingState(isLoading) {
+    const submitButton = this._popupForm.querySelector(".popup__button");
+
+    if (submitButton) {
+      if (isLoading) {
+        submitButton.textContent = "Saving...";
+      } else {
+        submitButton.textContent = this._submitButtonText;
+
+        // Close the modal when the loading state is set to false
+        this.close();
```

Comment:
Please, delete `this.close();` from line 43. It should be only in `index.js`


You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well